src/components/collect-button.tsx

```diff
@@ -16,33 +18,41 @@ interface IState {}
 export class CollectButtonComponent extends BaseComponent<IProps, IState> {
 
   public render() {
-    const { mouse } = this.props;
+    const { backpack } = this.stores;
+    const { backpackIndex } = this.props;
```

Comment:
Oh, one problem with just storing the index and not the mouse itself, and using this same view in your zoom model: If the user is looking at a mouse cell, and then deletes the mouse in the backpack, the next mouse down is now at that index, and that's what the circle over the cell shows.

We probably will eventually need some UI to prevent or discourage this, e.g. a confirmation dialog, but I think the naive response when this happens should be to clear the zoom row entirely.

Even if the mouse itself isn't the one that got deleted, the shifting of indices may still cause the zoom row to be showing the wrong mouse.